addon/create-class-computed.js

```diff
@@ -0,0 +1,93 @@
+import Ember from 'ember';
+import EmberObject from 'ember-object';
+import computed from 'ember-computed';
+import on from 'ember-evented/on';
+import observer from 'ember-metal/observer';
+import { isNone } from 'ember-utils';
+// import getOwner from 'ember-owner/get';
+import get from 'ember-metal/get';
+import { setProperties } from 'ember-metal/set';
+import WeakMap from 'ember-weakmap';
+import { mapKeysToValues } from './-build-computed';
+import getValue from './get-value';
+import collapseKeys from './collapse-keys';
+import flattenKeys from './flatten-keys';
+
+const { defineProperty } = Ember;
+
+const PROPERTIES = new WeakMap();
+
+function findOrCreatePropertyInstance(context, propertyClass, key) {
+  let propertiesForContext = PROPERTIES.get(context);
+  if (isNone(propertiesForContext)) {
+    propertiesForContext = {};
+    PROPERTIES.set(context, propertiesForContext);
+  }
+
+  let property = propertiesForContext[key];
+  if (property) {
+    return property;
+  }
+
+  // let owner = getOwner(context);
+  property = propertyClass.create(/*owner.ownerInjection(), */{
+    _key: key,
+    _context: context
+  });
+
+  propertiesForContext[key] = property;
+  return property;
+}
+
+const BaseClass = EmberObject.extend({
+  _computedDidChange: observer('_computed', function() {
+    this._context.notifyPropertyChange(this._key);
+  })
+});
+
+function createObserverClass(classKeysObj, macroGenerator) {
+  let classKeys = Object.keys(classKeysObj);
+  let observers = classKeys.filter(key => classKeysObj[key]);
+
+  function rewriteComputed() {
+    let values = mapKeysToValues(observers, getValue, this);
+    let cp = macroGenerator(...values);
+    defineProperty(this, '_computed', cp);
+  }
+
+  let classProperties = observers.reduce((properties, key) => {
+    properties[`${key}DidChange`] = observer(key, rewriteComputed);
+    return properties;
+  }, {});
+
+  return BaseClass.extend(classProperties, {
+    _onInit: on('init', rewriteComputed)
```

Comment:
> It's rare that I ever depend on _super call order

> I think calling `_super` is just always the first thing you do in an override (unless you have very specific reasons not to do so which would rarely be the case though).

> and it seems safer to attach on to events rather than overriding. Maybe I'm missing something?

I think the main point is that you can't be 100% sure exactly when the even will be triggered and you code be run which might lead to strange errors. I'm not 100% sure about all details of this but I think the reason is basically timing.